source/plugin/commands/argumentparsing.rst

```diff
@@ -0,0 +1,176 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. list-table::
+    :widths: 25 50 25
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+      - Value Type & Amount
+    * - ``string``
+      - Require the argument to be a string.
+      - one ``String``
+    * - ``remainingJoinedStrings``
+      - Concatenates all remaining arguments separated by spaces (useful for message commands).
+      - one ``String``
+    * - ``bool``
+      - Require the argument to be a boolean.
+      - one ``Boolean``
+    * - ``integer``
+      - Require the argument to be a integer.
+      - one ``Integer``
+    * - ``rangeInteger``
+      - Require the argument to be a integer between two values.
+      - one ``Integer``
+    * - ``doubleNumber``
+      - Require the argument to be a double.
+      - one ``Double``
+    * - ``rangeDouble``
+      - Require the argument to be a double between two values
+      - one ``Double``
+    * - ``player``
+      - Require the argument to be a player. *May return multiple players!*
+      - multiple ``ServerPlayer`` instances
+    * - ``playerOrTarget``
+      - Like ``player``, but returns the sender of the command if no matching player was found.
+      - mutiple ``ServerPlayer`` instances
+    * - ``user``
+      - Require the argument to be a user. *May return multiple users!*
+      - multiple ``User`` instances
+    * - ``world``
+      - Require the argument to be a world (only loaded worlds)
+      - multiple ``ServerWorld`` instances
+    * - ``location``
+      - Require the argument to be a location
+      - one ``ServerLocation``
+    * - ``vector3d``
+      - Require the argument to be a vector
+      - one ``Vector3d``
+    * - ``rotation``
+      - Require the argument to be a vector, but returns the senders rotation if no value was specified
+      - one ``Vector3d``
+    * - ``registryElement``
+      - Require the argument to be the resoure key of one of the specified elements
+      - multiple matching elements of the specified registry type
+    * - ``choices``
+      - Returns an argument that allows selecting from a limited set of values
+      - one specified value
+    * - ``literal``
+      - Require the argument to match one of the specified literals
+      - one specified literal
+    * - ``enumValue``
+      - Require the argument to be a enum
+      - one ``Enum`` 
+    * - ``seq``
+      - Builds a sequence of commands
+      - inherited
+
+.. tip::
+    See the Javadocs for :javadoc:`Parameter` for more information
+
+.. warning::
+
+    Don't expect that a ``Parameter`` will only ever return a single value, 
+    a lot of them support multiple return values; some might support regular expressions or use command selector. 
+    This is intensional as it makes commands easier to use. Example ``/tell @a BanditPlayer has cookies!``.
+    
+Custom Parameter
+~~~~~~~~~~~~~~~~
+
+It is possible to create custom command elements. Example Vector2i. This is done though the 
```

Comment:
`It is possible to create custom command elements; for example, Vector2i`

Reply:
Updated